code/headers/motor_control.hpp

```diff
@@ -8,23 +8,23 @@ namespace r2d2::moving_platform {
 	 * This class exists so you can use the motor even without knowing what motor you have been provided with. 
 	 */
     class motor_control_c {
-    protected:
-		/**
-		 * this speed represents a throtle in percentages (%)
-		 * This means that 100 is max and forward, -100 is backwards
-		 */
+    private:
+        /**
+		    * this speed represents a throtle in percentages (%)
+		    * This means that 100 is max and forward, -100 is backwards
+		    */
         int8_t speed;
 
     public:
-		/**
-		 * the speed set must not exceed 100
-		 */
-        virtual void set_speed(const int8_t &_speed) = 0;
-		
-		/**
-		 * returns the speed value
-		 */
-        inline int8_t get_speed() const {
+        /**
+		    * the speed set must not exceed 100
+		    */
+        virtual void set_speed(int8_t new_speed) = 0;
```

Comment:
```suggestion
        virtual void set_speed(int8_t &new_speed) = 0;
```
new_speed can be passed by reference

Reply:
It is indeed necessary to use const & to compile. with ints this is slower